python/tests/hive/test_hive_tables.py

```diff
@@ -157,3 +157,21 @@ def test_create_tables(client, current_call, base_scan_schema, base_scan_partiti
     current_call.return_value = tbl[0].args[0].parameters['metadata_location']
 
     tables.load("test.test_123")
+
+
+@mock.patch("iceberg.hive.HiveTableOperations.refresh_from_metadata_location")
+@mock.patch("iceberg.hive.HiveTables.delete_file")
+@mock.patch("iceberg.hive.HiveTableOperations.current")
+@mock.patch("iceberg.hive.HiveTables.get_client")
+def test_drop_tables(client, metadata, delete_file, refresh_call, tmpdir):
```

Comment:
I think it would be good to test purge as well tbh, most of the code complexity in this change lives there.

Reply:
Will do

Reply:
Added a test with a bunch of mocks to test that part.

Reply:
Spent some time to add a second test, with mocking all the method calls to test the new code.